django/contrib/postgres/fields/ranges.py

```diff
@@ -332,6 +341,15 @@ class RangeEndsWith(models.Transform):
     def output_field(self):
         return self.lhs.output_field.base_field
 
+    def as_postgresql(self, compiler, connection):
+        sql, params = self.as_sql(compiler, connection)
+        # Cast the lhs if needed.
+        cast_sql = ""
+        if isinstance(self.lhs, models.Expression) and self.lhs._output_field_or_none:
```

Comment:
Use `hasattr(self.lhs, "resolve_expression")` for checking that `self.lhs` is an expression.